examples/MvcExample/Controllers/HomeController.cs

```diff
@@ -84,34 +85,49 @@ public async Task<IActionResult> Complete([FromQuery(Name = "payment_id")]string
 
             var apiResponse = await _truelayer.Payments.GetPayment(paymentId);
 
-            IActionResult Failed(string status)
+            IActionResult Failed(string status, OneOf<Provider.UserSelected, Provider.Preselected>? providerSelection)
             {
                 ViewData["Status"] = status;
+
+                SetProviderAndSchemeId(providerSelection);
                 return View("Failed");
             }
 
-            IActionResult Success(PaymentDetails payment)
+            IActionResult Success(PaymentDetails payment, OneOf<Provider.UserSelected, Provider.Preselected> providerSelection)
```

Comment:
Just noticed that this (and Pending) take the `PaymentDetails` anyway, so you could just read the ProviderSelection from there.

Reply:
should be addressed in [86d5849](https://github.com/TrueLayer/truelayer-dotnet/pull/136/commits/86d5849a14c112ce1520ee1476c6d5d51f90297d)

Reply:
done in [1410b2e](https://github.com/TrueLayer/truelayer-dotnet/pull/136/commits/1410b2e64183058c017711716c75675470c9ff27)